src/controllers/assistant.js

```diff
@@ -13,34 +13,113 @@
 import {
   createResponse,
 } from '@adobe/spacecat-shared-http-utils';
+import { isObject } from '@adobe/spacecat-shared-utils';
+import { AssistantDto } from '../dto/assistant-response.js';
+import {
+  commandConfig,
+  getFirefallCompletion,
+  SCOPE,
+  validateAccessScopes,
+  STATUS,
+} from '../support/assistant-support.js';
 import { ErrorWithStatusCode } from '../support/utils.js';
 
 /**
  * Assistant controller. Provides methods to perform AI assisted operations.
+ * There are certain commands that can be executed by the assistant. Depending on the command
+ * there has to be certain inputs such as prompt and imageUrl.  A call is made to
+ * Firefall, and the response is returned.
  * @returns {object} Import assistant controller.
  * @constructor
  */
-function AssistantController() {
+function AssistantController(context) {
   const HEADER_ERROR = 'x-error';
+  const { auth, log } = context;
 
   function createErrorResponse(error) {
     return createResponse({}, error.status, {
       [HEADER_ERROR]: error.message,
     });
   }
 
+  function isBase64UrlImage(base64String) {
+    return base64String.startsWith('data:image/') && base64String.endsWith('=') && base64String.includes('base64');
+  }
+
+  function validateRequestData(data) {
+    const {
+      command, prompt, imageUrl,
+    } = data;
+
+    // Validate 'command'
+    if (!command) {
+      throw new ErrorWithStatusCode('Invalid request: command is required.', STATUS.BAD_REQUEST);
+    }
+    const currentCommandConfig = commandConfig[command];
+    if (!currentCommandConfig) {
+      throw new ErrorWithStatusCode(`Invalid request: command not implemented: ${command}`, STATUS.BAD_REQUEST);
+    }
+
+    if (currentCommandConfig.parameters.includes('prompt') && !prompt) {
+      throw new ErrorWithStatusCode('Invalid request: prompt is required.', STATUS.BAD_REQUEST);
+    }
+
+    if (currentCommandConfig.parameters.includes('imageUrl')) {
+      if (!imageUrl) {
+        throw new ErrorWithStatusCode('Invalid request: Image url is required.', STATUS.BAD_REQUEST);
+      }
+      // Only base64 images for now.
+      if (!isBase64UrlImage(imageUrl)) {
+        throw new ErrorWithStatusCode('Invalid request: Image url is not a base64 encoded image.', STATUS.BAD_REQUEST);
+      }
+    }
+  }
+
+  function parseRequestContext(requestContext) {
+    if (!requestContext || !isObject(requestContext)) {
+      throw new ErrorWithStatusCode('Invalid request: missing request context.', STATUS.BAD_REQUEST);
+    }
+    if (!requestContext.data || !requestContext.attributes) {
+      throw new ErrorWithStatusCode('Invalid request: invalid request context format.', STATUS.BAD_REQUEST);
+    }
+    const { data, attributes } = requestContext;
```

Comment:
```suggestion
    const { data: { command, prompt }, attributes } = requestContext;
```

Reply:
👍  I'll add `options` for the next line (86) as well.